pkg/build/api/types.go

```diff
@@ -76,8 +76,73 @@ type BuildSpec struct {
 	// scheduled in the system, that the build may be active on a node before the
 	// system actively tries to terminate the build; value must be positive integer
 	CompletionDeadlineSeconds *int64
+
+	// Info is a placeholder that can store multiple reasons why a build was
+	// triggered. It should cover for a future implementation of level triggered
+	// builds where you can multiple reasons why a build was triggered.
+	Info []*BuildTriggerInfo
```

Comment:
I'm placing a hold on this API change until I have more time to review.  We may be moving triggers up and out of objects to top level concepts and I want to have time to integrate this idea.  This was not intended for 1.2, was it?